operators/src/opencl/cl_program.rs

```diff
@@ -491,21 +553,63 @@ impl<'a> CLProgramRunnable<'a> {
             let geo_buffers = if argument.include_geo {
                 Some(match features.output_type {
                     VectorDataType::MultiPoint => {
-                        let coords = Buffer::<Double2>::builder()
-                            .queue(queue.clone())
-                            .len(features.num_coords())
-                            .build()?;
-                        kernel.set_arg(format!("OUT_POINT_COORDS{}", idx), &coords)?;
-
-                        let offsets = Buffer::<i32>::builder()
-                            .queue(queue.clone())
-                            .len(features.num_features() + 1)
-                            .build()?;
-                        kernel.set_arg(format!("OUT_POINT_OFFSETS{}", idx), &offsets)?;
-
-                        FeatureGeoOutputBuffer::Points(PointBuffers { coords, offsets })
+                        let coords = create_buffer::<Double2>(queue, features.num_coords())?;
+                        kernel.set_arg(format!("OUT_FEATURE{}_COORDS", idx), &coords)?;
+
+                        let feature_offsets =
+                            create_buffer::<i32>(queue, features.num_features() + 1)?;
+                        kernel.set_arg(format!("OUT_FEATURE{}_OFFSETS", idx), &feature_offsets)?;
```

Comment:
len parameters?

Reply:
Could you re-phrase this comment?

Reply:
I meant we need a len-parameter for each output array as well. If we check that we only read valid data from input arrays we should only write data to valid positions in the output array. Of course you can argue that its the caller responsibility to allocate enough memory but I'm not so sure. Additionaly maybe there are algorithms which are flexible in the output length, e.g. a histogram opencl kernel might need the information about the number of buckets which is equal to the length of the output array. Also it would be more uniform if all arrays come as a duo of pointer and length.

Reply:
e172d81